Generate the SDK manifest using Python

ebassi opened the pull request:
We're doing two passes to generate the SDK manifest: first from the
Makefile and then from the generate-manifest Python script. The script,
incidentally, overwrites some of the substitutions we perform from the
Makefile.

Let's move the manifest munging to the Python script only.

We need to still use the Makefile pattern substitution for various
ancillary files, so we need to use a different extension for the
manifest, to avoid Make getting confused.

ebassi committed:
Generate the SDK manifest using Python

We're doing two passes to generate the SDK manifest: first from the
Makefile and then from the generate-manifest Python script. The script,
incidentally, overwrites some of the substitutions we perform from the
Makefile.

Let's move the manifest munging to the Python script only.

We need to still use the Makefile pattern substitution for various
ancillary files, so we need to use a different extension for the
manifest, to avoid Make getting confused.

ptomato commented on generate-manifest.py:
```diff
@@ -44,7 +44,7 @@ def default_arch(debian=False):
     return arch
 
-def edit_manifest(data, arch, branch, runtime_version):
+def edit_manifest(data, arch, branch, runtime_version, fdo_version):
```
`fdo_version` doesn't seem to be used?

ptomato reviewed (changes requested):
- Is there anything that can be removed from the manifest template now? It might be a good idea to get rid of the substitutions that are just overwritten by the generate script.
- Needs a corresponding change in the `jenkins-flapjack-ci` script in eos-build, and in the example config file in flapjack.

ebassi commented:
This fell off my radar; I've rewritten the gtk3 patch handling a bit so it's not terrible, and dropped the `--fdo-runtime` option, as it was a leftover from a previous iteration.
